src/store.c

```diff
@@ -339,47 +407,104 @@ static void writer_close(
     close(store->fd);
 }
 
+static struct rill_pairs* flip_values(struct rill_pairs* pairs)
+{
+    struct rill_pairs* inverted = rill_pairs_new(pairs->cap);
+
+    for (size_t x = 0; x < pairs->cap; ++x) {
+        inverted->data[x].val = pairs->data[x].key;
+        inverted->data[x].key = pairs->data[x].val;
+    }
+
+    inverted->len = pairs->len;
+    inverted->cap = pairs->cap;
+
+    return inverted;
+}
+
 bool rill_store_write(
         const char *file,
-        rill_ts_t ts, size_t quant,
+        rill_ts_t ts,
+        size_t quant,
         struct rill_pairs *pairs)
 {
     rill_pairs_compact(pairs);
     if (!pairs->len) return true;
 
+    struct rill_pairs* inverted = flip_values(pairs);
```

Comment:
There is room for optimisation as we currently requires that we keep two uncompressed representation of the data in memory at the same time. This is going to cause problem when we try to transcode the existing database. We can deal with this later though.

Reply:
That was for a working first pass and this has been optimized now. By optimized, I mean I'm reusing your algorithm for the second column as well.